src/input/keyboard/mod.rs

```diff
@@ -580,6 +580,32 @@ impl<D: SeatHandler + 'static> KeyboardHandle<D> {
         );
     }
 
+    /// Iterate over the keysyms of the currently pressed keys.
+    pub fn with_pressed_keysyms<F, R>(&self, f: F)
+    where
+        F: FnOnce(Vec<KeysymHandle<'_>>) -> R,
+        R: 'static,
+    {
+        let guard = self.arc.internal.lock().unwrap();
+        {
+            let handles = guard
+                .pressed_keys
+                .iter()
+                .map(|code| KeysymHandle {
+                    keycode: code + 8,
+                    state: &guard.state,
+                    keymap: &guard.keymap,
+                })
+                .collect::<Vec<_>>();
+            f(handles);
+        }
+    }
+
+    /// Get the current modifiers state
+    pub fn modifier_state(&self) -> ModifiersState {
+        self.arc.internal.lock().unwrap().mods_state
+    }
```

Comment:
Could we get this one for Inner flavor of handle as well?

Reply:
Sure!